Makefile

```diff
@@ -10,19 +10,26 @@ BUILDFLAGS := -tags "$(BUILDTAGS)"
 
 PACKAGES := $(shell go list ./... | grep -v vendor)
 
-all: deps .gitvalidation test validate
+all: deps.timestamp .gitvalidation test validate
 
-deps: vndr
+vndr:
+	@go get -u github.com/LK4D4/vndr
+
+deps: vndr vendor deps.timestamp
 	go get -u $(BUILDFLAGS) github.com/golang/lint/golint
 	go get $(BUILDFLAGS) github.com/vbatts/git-validation
 
-vndr: vendor.conf
-	@go get -u github.com/LK4D4/vndr && vndr
+deps.timestamp:
+	@touch deps.timestamp
+
+vendor: deps.timestamp vendor.conf
+	@vndr
+	@touch vendor
 
 clean:
-	rm -rf vendor
+	rm -rf vendor deps.timestamp
 
-test: deps
+test: deps.timestamp
```

Comment:
How are these rules supposed to work? AFAICS all this does is that it causes the `deps.timestamp` rule to trigger, which just touches the timestamp file; under _no circumstances_ that would cause `deps` to be re-run AFAICS. Same for the `all` dependency.

It does seem that something like the `vendor:… vendor.conf; touch vendor` rule would be useful to ensure `vndr` is run only when necessary, but it’s not clear to me how all the pieces in this commit are supposed to fit together.